### PR TITLE
Add anchor support to dynamic blocks

### PR DESCRIPTION
**Pull request by Soean**
## Description
Add anchor support to dynamic blocks

## ToDo
- [ ] add Tests

### CODE REVIEW

File: packages/block-editor/src/hooks/anchor.js [hunk 1]
```diff
@@ -39,14 +39,20 @@ export function addAttribute( settings ) {
 	}
 	if ( hasBlockSupport( settings, 'anchor' ) ) {
 		// Gracefully handle if settings.attributes is undefined.
-		settings.attributes = {
-			...settings.attributes,
-			anchor: {
+		let anchor = {
+			type: 'string',
```
Comment (Soean) [inline packages/block-editor/src/hooks/anchor.js:43]: I am not sure why we need this line, this should already be added with PHP.

**Comment by gwwar** [comment]
@Soean were you still interested in working on this one?

**Comment by Jazz-Man** [comment]
@Soean This feature is quite interesting, so if you can, please do it:). I implemented it in my project based on your PR. But I would like to have it in WP core.

**Comment by talldan** [comment]
> I don't want to come across as snarky, but what caused this to be closed yesterday instead of merged 2 years ago, @talldan?

@cr0ybot You can't see it now, but there were about many merge conflicts to resolve and the code was untouched for a long time, so it was impossible to review correctly or merge. Also @Soean hadn't replied in over 1 1/2 years to the message above (https://github.com/WordPress/gutenberg/pull/24699#issuecomment-816249478).

The best thing to do would be to make a new pull request with conflicts resolved.

One of the problems with leaving a pull request open for a long time is it can also deter other contributors who are sensitive about stepping on someone else's toes, so I decided it's best to clear the decks. Someone had also assigned themselves to https://github.com/WordPress/gutenberg/issues/29401, and then not produced any work, so I've also removed that assignee, and hopefully now it's clear that this is a task that's available to pick up. I've also just added the 'Needs Dev' label to make it even clearer.

You are right though, I should've left a comment explaining that when I closed it. 👍

**Comment by Soean** [comment]
I will create a new pull request to solve this issue.

**Comment by Soean** [comment]
New WIP PR: #44771

**Pull request closed without merging**